### PR TITLE
Add New Model3D Guide

### PR DESCRIPTION
**Pull request by dawoodkhan82**
# Description
Add guide on using Model3D component

Please include: 
* relevant motivation
* a summary of the change 
* which issue is fixed. 
* any additional dependencies that are required for this change.

Closes: # (issue)
#1758

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: guides/how_to_use_3D_model_component.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+Docs: model3D
+
+## Introduction
+
+3D models are becoming more popular in machine learning and make for some of the most fun demos to experiment with. Using `gradio`, you can easily build a demo of your 3D image model and share it with anyone. The Gradio 3D Model component accepts 3 file types including: *.obj*, *.glb*, & *.gltf*.
+
+This guide will show you how to build a demo for your 3D image model in a few lines of code; like the one below. Play around with 3D object by clicking around, dragging and zooming:
+
+<iframe src="https://hf.space/embed/dawood/Model3D/+" frameBorder="0" height="350" title="Gradio app" class="container p-0 flex-grow space-iframe" allow="accelerometer; ambient-light-sensor; autoplay; battery; camera; document-domain; encrypted-media; fullscreen; geolocation; gyroscope; layout-animations; legacy-image-formats; magnetometer; microphone; midi; oversized-images; payment; picture-in-picture; publickey-credentials-get; sync-xhr; usb; vr ; wake-lock; xr-spatial-tracking" sandbox="allow-forms allow-modals allow-popups allow-popups-to-escape-sandbox allow-same-origin allow-scripts allow-downloads"></iframe>
+
```
Comment (dawoodkhan82) [inline guides/how_to_use_3D_model_component.md:14]: This is a basic model3d demo as of now. There isn't any models on the hub I can easily use to create a demo around.

File: guides/how_to_use_3D_model_component.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+Below is a demo that uses the DPT model to predict the depth of an image and then uses 3D Point Cloud to create a 3D object. Take a look at the [app.py](https://huggingface.co/spaces/radames/dpt-depth-estimation-3d-obj/blob/main/app.py) file for a peek into the code and the model prediction function.
+
+<iframe src="https://hf.space/embed/radames/dpt-depth-estimation-3d-obj/+" frameBorder="0" height="350" title="Gradio app" class="container p-0 flex-grow space-iframe" allow="accelerometer; ambient-light-sensor; autoplay; battery; camera; document-domain; encrypted-media; fullscreen; geolocation; gyroscope; layout-animations; legacy-image-formats; magnetometer; microphone; midi; oversized-images; payment; picture-in-picture; publickey-credentials-get; sync-xhr; usb; vr ; wake-lock; xr-spatial-tracking" sandbox="allow-forms allow-modals allow-popups allow-popups-to-escape-sandbox allow-same-origin allow-scripts allow-downloads"></iframe>
+
```
Comment (dawoodkhan82) [inline guides/how_to_use_3D_model_component.md:70]: Decided to add these to show off more complex 3d model demos.

File: guides/how_to_use_3D_model_component.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
+# How to Use the 3D Model Component
+
+Related spaces: https://huggingface.co/spaces/dawood/Model3D, https://huggingface.co/spaces/radames/PIFu-Clothed-Human-Digitization, https://huggingface.co/spaces/radames/dpt-depth-estimation-3d-obj
+Tags: VISION, IMAGE
+Docs: model3D
```
Comment (freddyaboulton) [inline guides/how_to_use_3D_model_component.md:5]: I think this needs to be `model3d` to correctly link with the docs.

File: guides/how_to_use_3D_model_component.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+## Introduction
+
+3D models are becoming more popular in machine learning and make for some of the most fun demos to experiment with. Using `gradio`, you can easily build a demo of your 3D image model and share it with anyone. The Gradio 3D Model component accepts 3 file types including: *.obj*, *.glb*, & *.gltf*.
+
+This guide will show you how to build a demo for your 3D image model in a few lines of code; like the one below. Play around with 3D object by clicking around, dragging and zooming:
+
+<iframe src="https://hf.space/embed/dawood/Model3D/+" frameBorder="0" height="350" title="Gradio app" class="container p-0 flex-grow space-iframe" allow="accelerometer; ambient-light-sensor; autoplay; battery; camera; document-domain; encrypted-media; fullscreen; geolocation; gyroscope; layout-animations; legacy-image-formats; magnetometer; microphone; midi; oversized-images; payment; picture-in-picture; publickey-credentials-get; sync-xhr; usb; vr ; wake-lock; xr-spatial-tracking" sandbox="allow-forms allow-modals allow-popups allow-popups-to-escape-sandbox allow-same-origin allow-scripts allow-downloads"></iframe>
```
Comment (freddyaboulton) [inline guides/how_to_use_3D_model_component.md:13]: nit: I think we're trying to use `<gradio-app>` from now on.

File: guides/how_to_use_3D_model_component.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+<iframe src="https://hf.space/embed/radames/dpt-depth-estimation-3d-obj/+" frameBorder="0" height="350" title="Gradio app" class="container p-0 flex-grow space-iframe" allow="accelerometer; ambient-light-sensor; autoplay; battery; camera; document-domain; encrypted-media; fullscreen; geolocation; gyroscope; layout-animations; legacy-image-formats; magnetometer; microphone; midi; oversized-images; payment; picture-in-picture; publickey-credentials-get; sync-xhr; usb; vr ; wake-lock; xr-spatial-tracking" sandbox="allow-forms allow-modals allow-popups allow-popups-to-escape-sandbox allow-same-origin allow-scripts allow-downloads"></iframe>
+
+
+----------
+
+And you're done! That's all the code you need to build an interface for your Model3D model. Here are some references that you may find useful:
+
+* Gradio's ["Getting Started" guide](https://gradio.app/getting_started/)
+* The final [chatbot demo](https://huggingface.co/spaces/dawood/Model3D) and [complete code](https://huggingface.co/spaces/dawood/Model3D/tree/main) (on Hugging Face Spaces)
```
Comment (freddyaboulton) [inline guides/how_to_use_3D_model_component.md:77]: typo: [chatbot demo]?

File: guides/how_to_use_3D_model_component.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+* `cache_examples`: saves the predicted output for the examples, to save time on inference.
+
+
+## Exploring mode complex Model3D Demos:
+
+Below is a demo that uses the DPT model to predict the depth of an image and then uses 3D Point Cloud to create a 3D object. Take a look at the [app.py](https://huggingface.co/spaces/radames/dpt-depth-estimation-3d-obj/blob/main/app.py) file for a peek into the code and the model prediction function.
+
+<iframe src="https://hf.space/embed/radames/PIFu-Clothed-Human-Digitization/+" frameBorder="0" height="350" title="Gradio app" class="container p-0 flex-grow space-iframe" allow="accelerometer; ambient-light-sensor; autoplay; battery; camera; document-domain; encrypted-media; fullscreen; geolocation; gyroscope; layout-animations; legacy-image-formats; magnetometer; microphone; midi; oversized-images; payment; picture-in-picture; publickey-credentials-get; sync-xhr; usb; vr ; wake-lock; xr-spatial-tracking" sandbox="allow-forms allow-modals allow-popups allow-popups-to-escape-sandbox allow-same-origin allow-scripts allow-downloads"></iframe>
```
Comment (freddyaboulton) [inline guides/how_to_use_3D_model_component.md:65]: I think you posted the same demo twice?

File: guides/how_to_use_3D_model_component.md [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+)
+
+demo.launch()
+```
+
+Let's break down the code above:
+
+`load_mesh`: This is our 'prediction' function and for simplicity, this function will take in the 3D model mesh and return it.
+
+Creating the Interface:
```
Comment (freddyaboulton) [inline guides/how_to_use_3D_model_component.md:50]: Nice, this is very informative

**Review by freddyaboulton — APPROVED** [review]
@dawoodkhan82 This is awesome! Left some nits/typo alerts but otherwise looks good to merge to me!

File: guides/how_to_use_3D_model_component.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+Below is a demo that uses the PIFu model to convert an image of a clothed human into a 3D digitized model. Take a look at the [spaces.py](https://huggingface.co/spaces/radames/PIFu-Clothed-Human-Digitization/blob/main/PIFu/spaces.py) file for a peek into the code and the model prediction function.
+
+<gradio-app space="radames/PIFu-Clothed-Human-Digitization"> </gradio-app>
+
+----------
+
+And you're done! That's all the code you need to build an interface for your Model3D model. Here are some references that you may find useful:
+
+* Gradio's ["Getting Started" guide](https://gradio.app/getting_started/)
+* The final [3D Model Demo](https://huggingface.co/spaces/dawood/Model3D) and [complete code](https://huggingface.co/spaces/dawood/Model3D/tree/main) (on Hugging Face Spaces)
```
Comment (abidlabs) [inline guides/how_to_use_3D_model_component.md:75]:
```suggestion
* The first [3D Model Demo](https://huggingface.co/spaces/dawood/Model3D) and [complete code](https://huggingface.co/spaces/dawood/Model3D/tree/main) (on Hugging Face Spaces)
```

**Review by abidlabs — APPROVED** [review]

**Comment by abidlabs** [comment]
Very nice!

**Pull request merged**